src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -4141,24 +4142,26 @@ theorem toInt_twoPow {w i : Nat} :
 theorem toInt_mul_toInt_lt {x y : BitVec w} : x.toInt * y.toInt ≤ 2 ^ (w * 2 - 2) := by
   rcases w with _|w
   · simp [of_length_zero]
-  · have xlt := toInt_lt (x := x); have xle := le_toInt (x := x)
-    have ylt := toInt_lt (x := y); have yle := le_toInt (x := y)
-    have h : 2 ^ ((w + 1) * 2 - 2) = 2 ^ ((w + 1) - 1) * 2 ^ ((w + 1) - 1) := by
-      rw [← Nat.pow_add, ←Nat.mul_two, Nat.mul_comm (m := 2) (n := (w + 1) - 1), Nat.mul_sub_one, Nat.mul_comm]
-    rw_mod_cast [h]
-    rw [← Nat.two_pow_pred_mul_two (by omega), Int.natCast_mul] at xlt ylt xle yle
-    exact Int.mul_le_mul_self (by omega) (by omega)
+  sorry
+  -- · have xlt := toInt_lt (x := x); have xle := le_toInt (x := x)
+  --   have ylt := toInt_lt (x := y); have yle := le_toInt (x := y)
+  --   have h : 2 ^ ((w + 1) * 2 - 2) = 2 ^ ((w + 1) - 1) * 2 ^ ((w + 1) - 1) := by
+  --     rw [← Nat.pow_add, ←Nat.mul_two, Nat.mul_comm (m := 2) (n := (w + 1) - 1), Nat.mul_sub_one, Nat.mul_comm]
+  --   rw_mod_cast [h]
+  --   rw [← Nat.two_pow_pred_mul_two (by omega), Int.natCast_mul] at xlt ylt xle yle
+  --   exact Int.mul_le_mul_self (by omega) (by omega)
 
 theorem le_toInt_mul_toInt {x y : BitVec w} : - (2 ^ (w * 2 - 2)) ≤ x.toInt * y.toInt := by
   rcases w with _|w
   · simp [of_length_zero]
-  · have xlt := toInt_lt (x := x); have xle := le_toInt (x := x)
-    have ylt := toInt_lt (x := y); have yle := le_toInt (x := y)
-    have h : 2 ^ ((w + 1) * 2 - 2) = 2 ^ ((w + 1) - 1) * 2 ^ ((w + 1) - 1) := by
-      rw [← Nat.pow_add, ←Nat.mul_two, Nat.mul_comm (m := 2) (n := (w + 1) - 1), Nat.mul_sub_one, Nat.mul_comm]
-    rw_mod_cast [h]
-    rw [← Nat.two_pow_pred_mul_two (by omega), Int.natCast_mul] at xlt ylt xle yle
-    exact Int.neg_mul_self_le_mul (by omega) (by omega) (by omega) (by omega)
+  · sorry
+    -- have xlt := toInt_lt (x := x); have xle := le_toInt (x := x)
```

Comment:
Luisa, try two_mul_toInt_lt. Try this PR: https://github.com/leanprover/lean4/commit/e7e57d40c4c5ee692a48a60a023327cc17c9bf83.

Reply:
thank you - I'm still at a stage where I'm sorrying things after merging master to understand what's broken and what happened over the past two weeks!

Reply:
OK. You should not need to "redo" proofs. If a broof is broken and you cannot figure it out in 10' let me know and we look at it together. This should be a trivial merge, if it is not lets first check why before starting to build new proofs.